payjoin/src/ohttp.rs

```diff
@@ -59,6 +59,26 @@ pub enum DirectoryResponseError {
     UnexpectedStatusCode(http::StatusCode),
 }
 
+impl PartialEq for DirectoryResponseError {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (DirectoryResponseError::InvalidSize(s1), DirectoryResponseError::InvalidSize(s2)) =>
+                s1 == s2,
+            (
+                DirectoryResponseError::OhttpDecapsulation(_),
+                DirectoryResponseError::OhttpDecapsulation(_),
+            ) => true,
+            (
+                DirectoryResponseError::UnexpectedStatusCode(s1),
+                DirectoryResponseError::UnexpectedStatusCode(s2),
+            ) => s1 == s2,
+            _ => false,
+        }
+    }
+}
+
+impl Eq for DirectoryResponseError {}
```

Comment:
It seems to me that the way to test error matching in cases where inner errors are not themselves `PartialEq` is to use `match`, and implementing `PartialEq` for these types is simply not possible because getting a completely correct implementation is not available without being able to compare the underlying types.